### PR TITLE
727 check for update feature

### PR DESCRIPTION
**Pull request by thomas-senechal**

### CODE REVIEW

File: int/systray/update/update_check.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+// We only cares about the 'tag_name' field, so we create a struct with only that field.
+type release struct {
+	TagName string `json:"tag_name"`
+}
+
+// Gets the latest version from GitHub.
+//
+// Performs a GET request to the GitHub API and decodes the response into a release struct.
+func getLatestVersion() (*version.Version, error) {
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, updateCheckURL, strings.NewReader(""))
```
Comment (peterjah) [inline int/systray/update/update_check.go:42]: why NewRequestWithContext is used for ?? compared to a simple http.get(...) ?

File: int/systray/update/update_check.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	systrayMenu.Refresh()
+}
+
+// Checks from the latest release on GitHub if there is a newer version available.
+func updateCheck(app *fyne.App, systrayMenu *fyne.Menu) {
+	log.Println("Checking for updates...")
+
+	latestVersion, err := getLatestVersion()
+	if err != nil {
+		log.Println("Error checking for updates:", err)
```
Comment (gregLibert) [inline int/systray/update/update_check.go:98]:
```suggestion
		log.Println("Error getting last version:", err)
```

File: int/systray/update/update_dialog.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	return window
+}
+
+// Creates the update dialog.
+func createUpdateDialog(window fyne.Window, app *fyne.App) dialog.Dialog {
+	dialog := dialog.NewCustomConfirm(
+		"An update is available for MassaStation. Do you want to update now?",
+		"Update",
+		"Cancel",
```
Comment (gregLibert) [inline int/systray/update/update_dialog.go:38]:
```suggestion
		"Ignore",
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: int/systray/update/update_dialog.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+// Finds the update dialog window by its title.
+func findUpdateWindow(app *fyne.App) fyne.Window {
+	var window fyne.Window
+
+	for _, w := range (*app).Driver().AllWindows() {
+		if w.Title() == updateWindowTitle {
+			window = w
+
+			break
```
Comment (gregLibert) [inline int/systray/update/update_dialog.go:26]:
```suggestion
			return w
```

File: int/systray/update/update_dialog.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		}
+	}
+
+	return window
+}
+
+// Creates the update dialog.
+func createUpdateDialog(window fyne.Window, app *fyne.App) dialog.Dialog {
+	dialog := dialog.NewCustomConfirm(
+		"An update is available for MassaStation. Do you want to update now?",
```
Comment (gregLibert) [inline int/systray/update/update_dialog.go:36]:
```suggestion
		"An update is available for MassaStation. Do you want to update it now?",
```

File: int/systray/update/update_dialog.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		"Update",
+		"Ignore",
```
Comment (gregLibert) [inline int/systray/update/update_dialog.go:38]:
```suggestion
		"Later",
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Comment by gregLibert** [comment]
@thomas-senechal can you resolve the conflicts?

**Pull request merged**